central/reports/common/query_builder_view_based.go

```diff
@@ -0,0 +1,76 @@
+package common
+
+import (
+	v1 "github.com/stackrox/rox/generated/api/v1"
+	"github.com/stackrox/rox/generated/storage"
+	"github.com/stackrox/rox/pkg/sac"
+	"github.com/stackrox/rox/pkg/sac/effectiveaccessscope"
+	"github.com/stackrox/rox/pkg/search"
+)
+
+// ReportQuery encapsulates the cve specific fields query, and the resource scope
+// queries to be used in a report generation run
+type ReportQueryViewBased struct {
+	CveFieldsQuery         string
+	DeploymentsScopedQuery *v1.Query
+}
+
+type queryBuilderViewBased struct {
+	vulnFilters *storage.ViewBasedVulnerabilityReportFilters
+}
+
+// NewVulnReportQueryBuilder builds a query builder to build scope and cve filtering queries for vuln reporting
+func NewVulnReportQueryBuilderViewBased(vulnFilters *storage.ViewBasedVulnerabilityReportFilters) *queryBuilderViewBased {
+	return &queryBuilderViewBased{
+		vulnFilters: vulnFilters,
+	}
+}
+
+// BuildQueryViewBased builds scope and cve filtering queries for view-based vuln reporting
+func (q *queryBuilderViewBased) BuildQueryViewBased(clusters []*storage.Cluster,
+	namespaces []*storage.NamespaceMetadata) (*ReportQueryViewBased, error) {
+	deploymentsScopedQuery, err := q.buildAccessScopeQueryViewBased(clusters, namespaces)
+	if err != nil {
+		return nil, err
+	}
+
+	cveQuery := q.buildCVEAttributesQueryViewBased()
+
+	return &ReportQueryViewBased{
+		CveFieldsQuery:         cveQuery,
+		DeploymentsScopedQuery: deploymentsScopedQuery,
+	}, nil
+}
+
+func (q *queryBuilderViewBased) buildCVEAttributesQueryViewBased() string {
+	vulnReportFilters := q.vulnFilters
+	return vulnReportFilters.GetQuery()
+}
+
+func (q *queryBuilderViewBased) buildAccessScopeQueryViewBased(clusters []*storage.Cluster,
```

Comment:
This looks to be basically the same as `buildAccessScopeQuery` in `query_builder.go`.  Recommend pulling it to `utils.go` and only having 1 of them.